src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -138,27 +128,39 @@ public void processResponseFiles(int batchSize) {
         }
     }
 
-    private void processInputUploadBlob(String inputUploadBlob) {
+    private void processInputUploadBlob(String inputUploadBlob, UserAccountEntity userAccount) {
         log.debug("Found ARM Input Upload file {}", inputUploadBlob);
         try {
             BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor = new BatchInputUploadFileFilenameProcessor(inputUploadBlob);
             String manifestName = generateManifestName(batchUploadFileFilenameProcessor.getUuidString());
+
+            String inputUploadFileRecordStr = armDataManagementApi.getBlobData(inputUploadBlob).toString();
+            log.info("Contents of ARM Input Upload file: '{}' '{}", inputUploadBlob, inputUploadFileRecordStr);
+            ArmResponseInputUploadFileRecord inputUploadFileRecord = objectMapper.readValue(inputUploadFileRecordStr, ArmResponseInputUploadFileRecord.class);
```

Comment:
You can't read the IU file like this. The IU file is not valid json. It should look something like this:
```
{"operation":"create_record","relation_id":"300072585","record_metadata":{"record_class":"DARTS","publisher":"DARTS","region":"GBR","recordDate":"2024-06-08T22:28:00Z","event_date":"2024-06-08T22:18:59Z","title":"annotation - File Size File Type InbountoUnstructured.docx","client_identifier":"300072585","contributor":"DMP-2247_Courthouse & Room1_DMP-2247","bf_001":"Annotation","bf_002":"DMP-2247_case4","bf_003":"application/vnd.openxmlformats-officedocument.wordprocessingml.document","bf_004":"2024-06-03T00:00:00Z","bf_005":"b7f83f6aada5f074615f337bbd6c5bfb","bf_009":"An annotation for Object documents checksum is null in object table DB","bf_010":"2024-06-08T22:18:59Z","bf_012":4545,"bf_013":4545,"bf_016":"-51","bf_019":"DMP-2247_Courthouse","bf_020":"Room1_DMP-2247"}}
{"operation":"upload_new_file","relation_id":"300072585","file_metadata":{"publisher":"DARTS","dz_file_name":"300072585_4545_1","file_tag":"docx"}}
{"operation":"create_record","relation_id":"300072586","record_metadata":{"record_class":"DARTS","publisher":"DARTS","region":"GBR","recordDate":"2024-06-08T22:28:00Z","event_date":"2024-06-08T22:18:53Z","title":"DMP-2247-file12","client_identifier":"300072586","contributor":"DMP-2247_Courthouse & Room1_DMP-2247","bf_001":"Media","bf_002":"DMP-2247_case4","bf_003":"mp2","bf_004":"2024-06-03T00:00:00Z","bf_005":"4d544cc2ad7a6f899c6703d29598ff89","bf_010":"2024-06-08T22:18:53Z","bf_011":"2024-06-03T10:14:00Z","bf_012":45249,"bf_013":45249,"bf_014":1,"bf_015":4,"bf_017":"2024-06-03T10:15:00Z","bf_019":"DMP-2247_Courthouse","bf_020":"Room1_DMP-2247"}}
{"operation":"upload_new_file","relation_id":"300072586","file_metadata":{"publisher":"DARTS","dz_file_name":"300072586_45249_1","file_tag":"mp2"}}
```
For each EOD, there will be 2 lines. One that represents create_record and one that represents the upload_new_file. For what you want you only want to read the create_record lines. You will need to iterate each line and only jsonify the create_record data. Each type of create_record has its own implementation so the media version is different to the transcription, case document and annotation document. These are represented by the classes that implement ArchiveRecordOperation. As you only need to get the relation id (EOD id) you probably wont need to use the specific versions.

Reply:
I spoke to @hemantasharma1129  and he confirmed the IU files in the stg environment are incorrect and provided me with a valid IU file that is a single JSON Object

Reply:
This is how IU file look like:
{"operation": "input_upload", "timestamp": "2024-01-11T12:46:21.215310", "status": 1, "exception_description": null, "error_status": null, "filename": "CGITestFilesMalformedManifest_1+2", "submission_folder": "/dropzone/A360/submission", "file_hash": "fbfec54925d62146aeced724ff9f3c8e"}

So the parsing logic should work with the new code.

Reply:
So what you will need is to catch the error if it can't read the json object and clean up the dangling files like so:

`private ArmResponseInputUploadFileRecord getResponseInputUploadOrDelete(BatchInputUploadFileFilenameProcessor batchInputUploadFileFilenameProcessor,
                                                                            String inputUploadFileRecordStr) throws IOException {
        try {
            return objectMapper.readValue(inputUploadFileRecordStr, ArmResponseInputUploadFileRecord.class);
        } catch (Exception e) {
            log.error("Unable to read ARM response input upload file {} - About to delete ",
                      batchInputUploadFileFilenameProcessor.getBatchMetadataFilenameAndPath(), e);
            deleteDanglingResponses(batchInputUploadFileFilenameProcessor);
            throw e;
        }
    }`

Reply:
@hemantasharma1129 do we also want to delete responses if the manifest file has invalid JSON (as currently this is only done if we can not find EODs for that file meaning or system has no knowledge of them) 
or in this scenario should our live support team contact ARM to send the corrected file as our system may have knowledge of data that "should" be in that file that we could not process due to the invalid JSON?

Reply:
@Ben-Edwards-cgi We should perform the error scenario part of separate ticket to avoid changing the scope of this ticket.
When the IU JSON is wrong we should do the following:
1. Find the associated response files and remove them. There are methods already to do so.
2. Find the associated EOD with IU file and tag them as failed in EOD table. Use the manifest_filename field of EOD to identify associated EOD.
3. Delete the IU file from ARM's response folder.